### PR TITLE
cider: set `meta.mainProgram` and `ozone` platform hints

### PR DESCRIPTION
**Pull request by MangoIV**
## Description of changes

- set the `mainProgram` for `cider` 
- add ozone platform hints (may require clearing the cache for users who have the `NIXOS_OZONE_WL` flag set)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by itslychee — APPROVED** [review]

**Comment by MangoIV** [comment]
this may require a cache clear if you were previously using cider with xwayland.

**Comment by itslychee** [comment]
It seemed to build & run fine without the flag wrapping, and I'm using sway as my window manager if that provides any context.

**Comment by MangoIV** [comment]
do you set the magic `NIXOS_OZONE_WL` flag? 
https://nixos.wiki/wiki/Wayland

**Comment by itslychee** [comment]
> do you set the magic `NIXOS_OZONE_WL` flag? 
> https://nixos.wiki/wiki/Wayland

I have not and it does not appear in my shell

**Comment by MangoIV** [comment]
yeah, then the wrapper script will not have effect and you will keep using xwayland. I cannot use xwayland because I have a HiDPI display and it looks awful... :D

**Comment by MangoIV** [comment]
yeah, then the wrapper script will not have effect and you will keep using xwayland. I cannot use xwayland because I have a HiDPI display and it looks awful... :D

**Comment by MangoIV** [comment]
cc @cigrainger

**Review by Mindavi — APPROVED** [review]
Bit weird that you have to source that script, but oh well.

**Comment by Mindavi** [comment]
Oh, I see this in the build log: substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file '/nix/store/bjqpql8rsgmlzzj53lbdilgns06956rj-cider-1.6.2/share/applications/cider.desktop')

May as well address that now too.

**Comment by MangoIV** [comment]
> Bit weird that you have to source that script, but oh well.

Yeah that seems to be a quirk of the builder.

**Comment by MangoIV** [comment]
> May as well address that now too.

Will do

**Comment by MangoIV** [comment]
alright, I have changed from `--replace` to `--replace-warn`

**Comment by MangoIV** [comment]
@itslychee you fancy readding your review, perhaps? perhaps it will be seen quicker if I get the "approvals:2" flag :3

**Comment by itslychee** [comment]
sure, one moment

**Review by itslychee — APPROVED** [review]

File: pkgs/applications/audio/cider/default.nix [hunk 1]
```diff
@@ -26,5 +30,6 @@ appimageTools.wrapType2 rec {
     license = licenses.agpl3;
     maintainers = [ maintainers.cigrainger ];
     platforms = [ "x86_64-linux" ];
+    mainProgram = "cider";
```
Comment (drupol) [inline pkgs/applications/audio/cider/default.nix:33]: Notice how `meta` attributes are sorted. They are _usually_ (not all the time) sorted alphabetically, can you please keep this order when adding `mainProgram` ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/cider/default.nix (hunk 1 above)
Comment (MangoIV) [reply pkgs/applications/audio/cider/default.nix:33]: Oh well, will adjust 😆

Comment (MangoIV) [reply pkgs/applications/audio/cider/default.nix:33]: Should this be in some sort of contributing guideline?

Comment (drupol) [reply pkgs/applications/audio/cider/default.nix:33]: Not for that, it's just common good sense.

Comment (Aleksanaa) [reply pkgs/applications/audio/cider/default.nix:33]: Not a blocker. Also someone already done it, and the merge conflict should be solved.

**Review by drupol — APPROVED** [review]

**Comment by MangoIV** [comment]
amazing! <3

**Pull request merged**